R/check.R

```diff
@@ -19,17 +19,15 @@
 #' f(x = 1, y = 2, z = 3)
 #' f(x = 1, y = 2, 3, 4, 5)
 check_dots_used <- function(env = parent.frame()) {
-  exit_handler <- bquote(
-    on.exit({
-      .(check_dots)(environment())
-    }, add = TRUE)
-  )
   eval_bare(exit_handler, env)
-
   invisible()
 }
 
 check_dots <- function(env = parent.frame()) {
+  if (.Call(ellipsis_dots_used, env)) {
+    return(invisible())
+  }
+
   proms <- dots(env)
```

Comment:
Could we just check `length()`  here? That seems a bit simpler, and my intuition is that the performance should be similar.

Reply:
You mean like this?

```r
  proms <- dots(env)
  if (!length(proms)) {
    return(invisible())
  }
```

This would only speed up the empty dots case. With dots this gives:

```
# A tibble: 5 x 14
  expression     min    mean  median     max `itr/sec` mem_alloc  n_gc n_itr
  <chr>      <bch:t> <bch:t> <bch:t> <bch:t>     <dbl> <bch:byt> <dbl> <int>
1 a_nodispa…  1.35µs  2.25µs  1.85µs 97.93µs   444144.        0B     2  9998
2 b_dispatch  2.73µs   4.3µs  3.45µs  1.04ms   232627.    3.77MB     2  9998
3 c_after     5.42µs   6.7µs  6.06µs  1.45ms   149155.   34.45KB     4  9996
4 d_length   14.03µs 18.85µs 16.47µs  1.35ms    53038.   36.38KB     8  9992
5 e_before   13.73µs 19.37µs 16.11µs  1.42ms    51624.   25.41KB     8  9992
# … with 5 more variables: total_time <bch:tm>, result <list>,
#   memory <list>, time <list>, gc <list>
```